scout/server/blueprints/cases/views.py

```diff
@@ -127,6 +127,187 @@ def generate_csv(header, lines):
     return data
 
 
+@cases_bp.route('/<institute_id>/<case_name>/mme_matches', methods=['GET', 'POST'])
+@templated('cases/matchmaker.html')
+def matchmaker_matches(institute_id, case_name):
+    """Show all MatchMaker matches for a given case"""
+    # check that only authorized users can access MME patients matches
+    user_obj = store.user(current_user.email)
+    if 'mme_submitter' not in user_obj['roles']:
+        flash('unauthorized request', 'warning')
+        return redirect(request.referrer)
+    # Required params for getting matches from MME server:
+    mme_base_url = current_app.config.get('MME_URL')
+    mme_token = current_app.config.get('MME_TOKEN')
+    if not mme_base_url or not mme_token:
+        flash('An error occurred reading matchmaker connection parameters. Please check config file!', 'danger')
+        return redirect(request.referrer)
+    institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
+    data = controllers.mme_matches(case_obj, institute_obj, mme_base_url, mme_token)
+    if data and data.get('server_errors'):
+        flash('MatchMaker server returned error:{}'.format(data['server_errors']), 'danger')
+        return redirect(request.referrer)
+    elif not data:
+        data = {
+            'institute' : institute_obj,
+            'case' : case_obj
+        }
+    return data
+
+
+@cases_bp.route('/<institute_id>/<case_name>/mme_match/<target>', methods=['GET','POST'])
+def matchmaker_match(institute_id, case_name, target):
+    """Starts an internal match or a match against one or all MME external nodes"""
+    institute_obj, case_obj = institute_and_case(store, institute_id, case_name)
+
+    # check that only authorized users can run matches
+    user_obj = store.user(current_user.email)
+    if 'mme_submitter' not in user_obj['roles']:
+        flash('unauthorized request', 'warning')
+        return redirect(request.referrer)
+
+    # Required params for sending an add request to MME:
```

Comment:
Is it worth it making a member method for the recurring things here as well? E.g. the population and checks are re-used almost the same in all of the methods.

Reply:
You mean a method that collects these variables?
mme_base_url = current_app.config.get('MME_URL')
mme_accepts = current_app.config.get('MME_ACCEPTS')
mme_token = current_app.config.get('MME_TOKEN')

Reply:
Hm, yes, it looked like more. Still, the error checking simplifies into one call. One could make a single Mme object abstracting all of the options, but hey, we have to have something to refactor later.. 😉

Reply:
I agree!